bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/datalayer/DataLayerComponent.java

```diff
@@ -69,10 +74,14 @@ protected ComponentData getComponentData() {
     }
 
     protected String generateId() {
-        String resourceType = resource.getResourceType();
-        String prefix = StringUtils.substringAfterLast(resourceType, "/");
-        String path = resource.getPath();
-        return StringUtils.join(prefix, ID_SEPARATOR, StringUtils.substring(DigestUtils.sha256Hex(path), 0, 10));
+        if (wcmComponent == null) {
```

Comment:
how can `wcmComponent` be `null`?

We only have 2 models using an optional default injection strategy (button v1 and v2). For all others the `wcmComponent` is required.

Reply:
DataLayerComponent is also extended by classes which are not Sling models, we handle that case here.

Reply:
Can you add this detail to the javadoc of the class while you are on it?

Reply:
What detail?

Reply:
That this abstract class may be extended by classes that are annotated as sling models and others that are not.